docs/doc/60-contributing/03-rfcs/20221105-infer-schema.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Infer Schema
+---
+
+- RFC PR: [datafuselabs/databend#8645](https://github.com/datafuselabs/databend/pull/8645)
+- Tracking Issue: [datafuselabs/databend#8646](https://github.com/datafuselabs/databend/issues/8646)
+
+## Summary
+
+Allowing infer schema from staged files or locations.
+
+## Motivation
+
+To load data from a stage or location, users need to create table first. However, sometimes users don't know the file schema, or the schema is too complex / too simple to be input by users.
+
+Allowing infer schema from existing files makes our users' lives easier. Also, this feature will unlock the databend from implementing `select * from @my_stage`.
+
+## Guide-level explanation
+
+Users will be able to infer the schema of the staged file by:
+
+```sql
+INFER @my_stage/data.csv FILE_FORMAT = ( TYPE = CSV);
+
++-------------+---------+----------+
+| COLUMN_NAME | TYPE    | NULLABLE |
+|-------------+---------+----------|
+| CONTINENT   | TEXT    | True     |
+| COUNTRY     | VARIANT | True     |
++-------------+---------+----------+
+```
+
+Infer from an external location is also supported:
+
+```sql
+INFER 's3://mybucket/data.csv' FILE_FORMAT = ( TYPE = CSV );
+
++-------------+---------+----------+
+| COLUMN_NAME | TYPE    | NULLABLE |
+|-------------+---------+----------|
+| CONTINENT   | TEXT    | True     |
+| COUNTRY     | VARIANT | True     |
++-------------+---------+----------+
+```
+
+`CREATE TABLE` will support `CREATE TABLE <table> BY ( <Query> )` to accept the output from `INFER`, so users can create a table with `INFRE` directly.
```

Comment:
we already support `create table as`

```

MySQL [(none)]> drop table a;
Query OK, 0 rows affected (0.004 sec)

MySQL [(none)]> create table a as select 3 as a, '3' as b;
Query OK, 0 rows affected (0.006 sec)

MySQL [(none)]> desc a;
+-------+------------------+------+---------+-------+
| Field | Type             | Null | Default | Extra |
+-------+------------------+------+---------+-------+
| a     | TINYINT UNSIGNED | NO   | 0       |       |
| b     | VARCHAR          | NO   |         |       |
+-------+------------------+------+---------+-------+
2 rows in set (0.003 sec)
```

Reply:
https://github.com/datafuselabs/databend/pull/8645#discussion_r1014596724

I think create table as is different too.